apis/core/v1alpha2/function_conversion.go

```diff
@@ -360,7 +360,7 @@ func (dst *Function) convertServingFrom(src *v1beta1.Function) error {
 		}
 	}
 
-	if src.Spec.Serving.Annotations != nil {
+	if src.Spec.Serving.Annotations != nil && src.Spec.Serving.Runtime == v1beta1.Async {
```

Comment:
<img width="458" alt="1665622955(1)" src="https://user-images.githubusercontent.com/53003665/195475162-99eead3e-341b-4241-aac5-6591ae53f264.png">

I think this modification is more appropriate, because the annotations may include some annotations of knative, such as `serving.knative.dev/rollout-duration`.

In addition, I think the condition of `strings.HasPrefix(k, "dapr.io")` should be deleted.